src/Catalog/SortingCollector.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace NuGet.Services.Metadata.Catalog
 {
-    public abstract class SortingCollector : CommitCollector
+    public abstract class SortingCollector<T> : CommitCollector
```

Comment:
`IEquatable<T>` is the right thing. Proper implementations of `IEquatable<T>` override `GetHashCode`. See the **Notes to Implementers** on [MSDN](https://msdn.microsoft.com/en-us/library/ms131187(v=vs.110).aspx).